onnxruntime/core/providers/webgpu/webgpu_context.cc

```diff
@@ -616,6 +616,29 @@ void WebGpuContext::EndProfiling(TimePoint /* tp */, profiling::Events& events,
   }
 }
 
+void WebGpuContext::PushErrorScopeIfNoLowerThan(webgpu::ValidationMode validation_mode) {
```

Comment:
`PushErrorScopeIfNoLowerThan` is not easier to understand than just put an if-clause outside.

I think we can keep their names of `ComputeContext::PushErrorScope()` and `ComputeContext::PopErrorScope()`, but move the condition out to `WebGpuExecutionProvider::OnRunStart`/`WebGpuExecutionProvider::OnRunEnd` and `WebGpuKernel::Compute`

Reply:
Sounds fair, Done!